helpers/osinfo.go

```diff
@@ -0,0 +1,231 @@
+package helpers
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+type OSReleaseID uint32
+
+func (o OSReleaseID) String() string {
+	return osReleaseIDToString[o]
+}
+
+const (
+	UBUNTU OSReleaseID = iota + 1
+	FEDORA
+	ARCH
+	DEBIAN
+	CENTOS
+	STREAM
+	ALMA
+)
+
+// stringToOSReleaseID is a map of supported distributions
+var stringToOSReleaseID = map[string]OSReleaseID{
+	"ubuntu": UBUNTU,
+	"fedora": FEDORA,
+	"arch":   ARCH,
+	"debian": DEBIAN,
+	"centos": CENTOS,
+	"stream": STREAM,
+	"alma":   ALMA,
+}
+
+// osReleaseIDToString is a map of supported distributions
+var osReleaseIDToString = map[OSReleaseID]string{
+	UBUNTU: "ubuntu",
+	FEDORA: "fedora",
+	ARCH:   "arch",
+	DEBIAN: "debian",
+	CENTOS: "centos",
+	STREAM: "stream",
+	ALMA:   "alma",
+}
+
+const (
+	OS_NAME osReleaseField = iota + 0
+	OS_ID
+	OS_ID_LIKE
+	OS_PRETTY_NAME
+	OS_VARIANT
+	OS_VARIANT_ID
+	OS_VERSION
+	OS_VERSION_ID
+	OS_VERSION_CODENAME
+	OS_BUILD_ID
+	OS_IMAGE_ID
+	OS_IMAGE_VERSION
+	OS_KERNEL_RELEASE // not part of default os-release, but we can use it here to facilitate things
+)
+
+type osReleaseField uint32
```

Comment:
This still needs to be exported for people to be able to use the `.String()` method

Reply:
Yep, made a mistake. Just realized that now. Thanks.